daemon/src/wire.rs

```diff
@@ -619,43 +614,11 @@ mod tests {
         assert!(Version::is_compatible(&taker_version, &maker_version));
     }
 
-    #[test]
-    fn given_maker_patch_ahead_of_taker_then_versions_compatible() {
-        let maker_version = Version(semver::Version::new(2, 0, 1));
-        let taker_version = Version(semver::Version::new(2, 0, 0));
-
-        assert!(Version::is_compatible(&taker_version, &maker_version));
-    }
-
-    #[test]
-    fn given_maker_minor_and_patch_ahead_of_taker_then_versions_compatible() {
-        let maker_version = Version(semver::Version::new(2, 1, 1));
-        let taker_version = Version(semver::Version::new(2, 0, 0));
-
-        assert!(Version::is_compatible(&taker_version, &maker_version));
-    }
-
```

Comment:
I would probably keep these tests and modify the other ones! having them won't hurt, we're cementing explicit support for patch version changes :)